pkg/da/data_availability_header_test.go

```diff
@@ -0,0 +1,210 @@
+package da
+
+import (
+	"bytes"
+	"strings"
+	"testing"
+
+	"github.com/celestiaorg/celestia-core/pkg/consts"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestNilDataAvailabilityHeaderHashDoesntCrash(t *testing.T) {
+	// This follows RFC-6962, i.e. `echo -n '' | sha256sum`
+	var emptyBytes = []byte{0xe3, 0xb0, 0xc4, 0x42, 0x98, 0xfc, 0x1c, 0x14, 0x9a, 0xfb, 0xf4, 0xc8,
+		0x99, 0x6f, 0xb9, 0x24, 0x27, 0xae, 0x41, 0xe4, 0x64, 0x9b, 0x93, 0x4c, 0xa4, 0x95, 0x99, 0x1b,
+		0x78, 0x52, 0xb8, 0x55}
+
+	assert.Equal(t, emptyBytes, (*DataAvailabilityHeader)(nil).Hash())
+	assert.Equal(t, emptyBytes, new(DataAvailabilityHeader).Hash())
+}
+
+func TestMinDataAvailabilityHeader(t *testing.T) {
+	dah := MinDataAvailabilityHeader()
+	expectedHash := []byte{
+		79, 166, 58, 67, 196, 20, 183, 43, 65, 70, 164, 145, 174, 69, 0, 144,
+		36, 208, 33, 100, 182, 249, 86, 254, 17, 254, 214, 142, 123, 208, 159, 169,
+	}
+	require.Equal(t, expectedHash, dah.hash)
+	require.NoError(t, dah.ValidateBasic())
+	// important note: also see the types.TestEmptyBlockDataAvailabilityHeader test
+	// which ensures that empty block data results in the minimum data availability
+	// header
+}
+
+func TestNewDataAvailabilityHeader(t *testing.T) {
+	type test struct {
+		name         string
+		expectedHash []byte
+		expectedErr  bool
+		squareSize   uint64
+		shares       [][]byte
+	}
+
+	tests := []test{
+		{
+			name:        "typical",
+			expectedErr: false,
+			expectedHash: []byte{
+				0x6e, 0xf4, 0x94, 0x23, 0x0, 0x15, 0x8c, 0x12, 0x54, 0xca, 0xb, 0xcd, 0xe8, 0x5, 0x8c, 0xc6,
+				0xc8, 0x13, 0xf2, 0xea, 0xaa, 0x48, 0x24, 0x1b, 0x27, 0x34, 0xf5, 0x39, 0x66, 0x8, 0xdb, 0xa6,
+			},
+			squareSize: 2,
+			shares:     generateShares(4, 1),
+		},
+		{
+			name:        "max square size",
+			expectedErr: false,
+			expectedHash: []byte{
+				0xa1, 0xda, 0x0, 0x73, 0x2b, 0x55, 0x93, 0x9d, 0xf6, 0x91, 0xd0, 0xa6, 0x23, 0x7d, 0xf9, 0xd7,
+				0x52, 0x60, 0xce, 0x48, 0x37, 0xcc, 0x1, 0xd4, 0x25, 0x65, 0xe5, 0xa1, 0xcd, 0xd1, 0x1d, 0x89,
+			},
+			squareSize: consts.MaxSquareSize,
+			shares:     generateShares(consts.MaxSquareSize*consts.MaxSquareSize, 99),
+		},
```

Comment:
we will have to change this test everytime we change the max square size :man_shrugging: